iroha-cli/interactive/interactive_common_cli.hpp

```diff
@@ -46,8 +47,11 @@ namespace iroha_cli {
       RESULT
     };
 
+
+    // Parameter prompt and default/cache value pair
+    using ParamData = std::pair<std::string, std::string>;
     // Description of parameters
-    using ParamsDescription = std::vector<std::string>;
+    using ParamsDescription = std::vector<std::shared_ptr<ParamData>>;
```

Comment:
Why `shared_ptr`?

Reply:
I mean why not just `std::vector<ParamData>`?

Reply:
Because it gets copied and I lose the reference to the initial objection when trying to update the cache ([interactive_common_cli.cpp:148](https://github.com/hyperledger/iroha/blob/af57b48d245811ea48c61778833761e7eb5e5409/iroha-cli/interactive/impl/interactive_common_cli.cpp#L148)).

Reply:
Hmm, it looks useless. Parameter in function is `auto`, not `auto &`, so modification of `param` bring nothing

Reply:
I'll give another go at using `std::vector<ParamData>`. IIRC I couldn't get it to compile when tried to return a reference in [findInHandlerMap](https://github.com/hyperledger/iroha/blob/af57b48d245811ea48c61778833761e7eb5e5409/iroha-cli/interactive/interactive_common_cli.hpp#L187).

Reply:
boost::optional [can handle](http://coliru.stacked-crooked.com/a/b219f68e6a3c4981) lvalue refs. Try out [this](http://termbin.com/kzb6) diff